### ISSUE TITLE
✨ FIX: Add contributor "type" to contributors.yaml file in the workflow

### ISSUE THREAD

**Issue by lwasser** [pyOpenSci/update-web-metadata#8]
Currently, the workflow in this repo will create an output file [called contributors.yml that is a data file](https://github.com/pyOpenSci/pyopensci.github.io/blob/main/_data/contributors.yml). 

[This data file drives our websites contributors listing here. ](https://www.pyopensci.org/our-community/). The data file template for a single package is below. 

```
- name: Inessa Pawson
  advisory: true
  bio: "NumPy Contributor Experience Lead\r\nOpen Source Program Manager at OpenTeams\
      \ Incubator"
  organization: '@numpy'
  github_username: InessaPawson
  github_image_id: 43481325
  twitter: inessapawson
  mastodon:
  orcidid:
  website:
  contributor_type:
  packages-submitted:
    -
  packages-reviewed:
    -
  packages-editor:
    -
  location:
  email:  
```

In our cotributor page we have almost 100 people. it would be nice to be able to sort contributors by roles. So update the contributors workflow to update the contributor_type using the following rules

    - reviewer: if the name is listed in a github review as a reviewer (or has a package listed in reviews) - we can add reviewer to contrib type
    - editor - if the name is listed in a github review as editor (or has a package under editor) add editor
    - guidebook-reviewer: if the person's info came from either the python package guide json file or the peer review guide all contributors json file add as guidebook reviewer
    - leadership - if executive countil
    - advisory - if advisory=True in the yaml file

````
contributor_type:
    - reviewer
    - editor
    - guidebook-reviewer
    - leadership
    - advisory
 ```
there are actually a few ways that this could be done. but it's worth thinking through a workflow as right now the workflow is broken up into a few pieces. 

1. Contributors are updated by parsing .json files across all repos, checking to see if they are alraedy in the contributors.yml file and then added. During that step we could add contrib type to anyone who has contributed to software review or packaging guide

**Comment by meerkatters** [comment]
I’ll be working on this for the PyCon23  mentored sprints

**Comment by lwasser** [comment]
@all-contributors please add @meerkatters for code, review

**Comment by meerkatters** [comment]
Hey @lwasser , so great to work with you to make my first OS contribution! now to finish this issue...


I am seeing potentially 3 subtasks on this issue

1. Sorting the .xaml files by contributor_type, is this a correct understanding?

2. updating the contributor_type assignments as the appear in the .yaml.
   You shared some rules for how the contributor_types should be derived. iiuc, 
   the task is to update the code to reflect these rules.

3. bug: some contributors are not making it into the yaml. Yesterday you manually 
   added me via GitBot, i see myself on the UPDATE-WEB-METADATA json. My pull request
   yesterday added the UPDATE-WEB-METADATA repo to the list of jsons to parse, 
   however when i brought in the pull to my fork and run the parse_contributors.py
   i am not seeing myself populate in the yaml, so i believe my pull was an incomplete
   update and more work is needed to either fix/finish this feature.


I’m around today if you have time to chat in person, else I'll look forward to any 
clarification you can provide here whenever is convenient.

**Comment by lwasser** [comment]
hi @meerkatters !! so sorry for the slow reply today! conferences are so busy!
are you around tomorrow by chance? we decided to have a small sprint tomorrow given two of us will be here and are leaving on tuesday. if not i'm in the main hall now listening to the wrap up sessions and we could potentially chat after briefly?

**Comment by lwasser** [comment]
i am so happy that you attended our sprint yesterday and was so psyched that you were able to submit your FIRST PR!! so huge!! ✨  🚀

**Comment by meerkatters** [comment]
I’m here by the room!

**Comment by lwasser** [comment]
@meerkatters so glad that we were able to connect briefly yesterday!i *think* that this issue really is now about a single bug. The sorting is a nice feature but we actually don't need things sorted. The way the website works, it simply adds the tags to the users section in the html as classes and then will allow us to filter by user contrib type. In this specific case, order doesn't matter. 

However the issue now is that a user that has contributed to multiple repositories meaning they should be assigned multiple types, is only getting 1/x types that should be assigned to them. 


David Nicholson for instance - 

```
- name: David Nicholson
  sort: 2
  title: Editor in Chief
  editorial-board: true
  bio: ML, AI; behavior + cog + neuro. Open + inclusive science. Pythonista. He/him/they/them.
      Habla espanglish y baila salsa y bachata a medio tiempo.
  organization:
  github_username: NickleDave
  github_image_id: 11934090
  twitter: nicholdav
  mastodon: https://fosstodon.org/@nicholdav
  orcidid: 0000-0002-4261-4719
  website: https://nicholdav.info/
  contributor_type:
    - current editor
    - contributor
    - web-contrib
```

David is our editor in chief as such should have peer-review and package guide. His name is in the [peer review](https://github.com/pyOpenSci/software-peer-review#contributors-) and packaging guide readme files as a contrib there but he is only listed as being a web-contrib. somehow it's only adding ONE contribution type when it parses each user

**Comment by lwasser** [comment]
i hope that makes sense and if it doesn't please ask additional questions so i can better clarify. I think we just need to troubleshoot the bug at this point. (but please tell me if you think i'm wrong!)

**Comment by lwasser** [comment]
following up here! i just fixed this via #42 @meerkatters if you ever want to work on something else please let me know! OR perhaps i'll see you at pycon next year!!?? and we can sprint together there :) many thanks for all of the help ! it was wonderful meeting you at pycon and i hope to run into you again!
